src/hidimstat/dcrt.py

```diff
@@ -1,307 +1,410 @@
+        Parameters for X distillation Lasso. If None, uses params_lasso_screening.
+    params_lasso_distillation_y : dict
+        Parameters for y distillation Lasso. If None, uses params_lasso_screening.
     refit : bool, default=False
-        Whether to refit on estimated support set
+        Whether to refit model on selected features
     screening : bool, default=True
-        Whether to screen variables
+        Whether to perform variable screening step. This selection is based on the coefficients of a cross-fitted Lasso model, keeping only the fraction (defined by screening_threshold) of variables with the largest coefficient and assigning a p-value of 1 to all other covariates.
     screening_threshold : float, default=0.1
-        Threshold for variable screening (0-100)
+        Percentile threshold for screening (0-100)
     statistic : {'residual', 'random_forest'}, default='residual'
-        Learning method for outcome distillation
+        Method for computing test statistics. Options are:
+        - 'residual': Uses Lasso regression residuals to compute test statistics.
+          This is faster and theoretically supported for linear relationships.
+        - 'random_forest': Uses Random Forest predictions to compute test statistics.
+          This can capture non-linear relationships but is computationally more intensive.
     centered : bool, default=True
-        Whether to standardize features
+        Whether to center and scale features
     n_jobs : int, default=1
         Number of parallel jobs
     joblib_verbose : int, default=0
-        Verbosity level
+        Verbosity level for parallel jobs
     fit_y : bool, default=False
         Whether to fit y using selected features
     n_tree : int, default=100
         Number of trees for random forest
     problem_type : {'regression', 'classification'}, default='regression'
-        Type of learning problem
+        Type of prediction problem
     random_state : int, default=2022
-        Random seed
+        Random seed for reproducibility
 
-    Returns
-    -------
+    Attributes
+    ----------
     selection_features : ndarray of shape (n_features,)
-        Boolean mask of selected features
-    X_res : ndarray of shape (n_selected, n_samples)
-        Residuals after X distillation
-    sigma2 : ndarray of shape (n_selected,)
+        Boolean mask indicating selected features
+    X_residual : ndarray
+        Residuals from X distillation
+    sigma2 : ndarray
         Estimated residual variances
-    y_res : ndarray of shape (n_selected, n_samples)
-        Response residuals
+    y_residual : ndarray
+        Residuals from y distillation
+    ts : ndarray of shape (n_features,)
+        test statistics following a standard normal distribution for all features
+    coef_X_full : ndarray
+        Estimated feature coefficients
+    clf_screening : estimator
+        Fitted screening model
+    clf_x_residual : array of estimators
```

Comment:
It is really weird to manipulate arrays of estimators. Why not list of estimatores ?

Reply:
it's  a list not an array. Sorry for the confusion.